ydb/core/tx/schemeshard/schemeshard__operation_side_effects.cpp

```diff
@@ -750,6 +771,89 @@ void TSideEffects::DoPersistDeleteShards(TSchemeShard *ss, NTabletFlatExecutor::
     ss->PersistShardsToDelete(db, ToDeleteShards);
 }
 
+void TSideEffects::DoUpdateTempTablesToCreateState(TSchemeShard* ss, const TActorContext &ctx) {
+    for (auto& [ownerActorId, tempTables]: TempTablesToCreateState) {
+
+        auto& tempTablesByOwner = ss->TempTablesState.TempTablesByOwner;
+        auto& nodeStates = ss->TempTablesState.NodeStates;
+
+        auto it = tempTablesByOwner.find(ownerActorId);
+
+        auto nodeId = ownerActorId.NodeId();
+
+        auto itNodeStates = nodeStates.find(nodeId);
+        if (itNodeStates == nodeStates.end()) {
+            auto& nodeState = nodeStates[nodeId];
+            nodeState.Owners.insert(ownerActorId);
+            nodeState.RetryState.CurrentDelay =
+                TDuration::MilliSeconds(ss->BackgroundCleaningRetrySettings.GetStartDelayMs());
+        } else {
+            itNodeStates->second.Owners.insert(ownerActorId);
+        }
+
+        if (it == tempTablesByOwner.end()) {
+            ctx.Send(new IEventHandle(ownerActorId, ss->SelfId(),
```

Comment:
I don't particularly like the idea to remember all session actors and probe node existence for every session individually.
All that temp table background cleaning required is to get notified about node disconnects -- disregard of actual number of sessions on that node and their specifics. And node existence probe could have used any actor (or may be service(?)) on that node, it doesn't have to be specifically session actor. 

May be there is a way to achieve that -- to remember, store and monitor nodes instead of all session actors?